OpenEdXMobile/src/main/java/org/edx/mobile/model/course/CourseComponent.java

```diff
@@ -176,6 +179,59 @@ public void setBlockCount(BlockCount count) {
         this.blockCount = blockCount;
     }
 
+    @Override
+    public boolean isCompleted() {
+        return this.completion == 1 || isAllChildCompleted();
+    }
+
+    @Override
+    public void setCompleted(int completion) {
+        this.completion = completion;
+        if (getType() != BlockType.DISCUSSION) {
+            setCompletionForNonDiscussionComponents();
+        }
+    }
+
+    private void setCompletionForNonDiscussionComponents() {
+        boolean isAllSiblingsCompleted = true;
+        if (getParent().getChildren().size() > 0) {
+            for (IBlock iBlock : getParent().getChildren()) {
+                if (iBlock.getType() != BlockType.DISCUSSION && !iBlock.isCompleted()) {
+                    isAllSiblingsCompleted = false;
+                }
+            }
+        }
+        if (isAllSiblingsCompleted) {
+            for (IBlock iBlock : getParent().getChildren()) {
+                if (iBlock.getType() == BlockType.DISCUSSION) {
+                    iBlock.setCompleted(1);
+                }
+            }
+        }
+    }
```

Comment:
please share the use case of why we need this method.

Reply:
Also, deserve doc strings.

Reply:
Done